src/lib/watcher/watcher.ts

```diff
@@ -180,22 +179,25 @@ export async function createWatcher(options: Options): Promise<Watcher> {
     }
     restarting = true
     clearConsole()
-    log.trace('hook', { name: 'beforeWatcherStartOrRestart' })
+    log.info('restarting', change)
 
+    log.trace('hook', { name: 'beforeWatcherStartOrRestart' })
     for (const plugin of plugins) {
       const runnerChanges = await plugin.dev.onBeforeWatcherStartOrRestart?.(change)
       if (runnerChanges) {
         link.updateOptions(runnerChanges)
       }
     }
+
     log.trace('hook', { name: 'beforeWatcherRestart' })
     plugins.forEach((p) => {
       p.dev.onBeforeWatcherRestart?.()
     })
-    log.info('restarting', change)
+
     options.events?.({ type: 'restart', file: change.file, reason: change.type })
     await link.startOrRestart()
     restarting = false
+    watcher.resume()
```

Comment:
We used to resume the watcher when the server sent the "server ready" signal, but with reflection that's no longer possible as there's no server running anymore (and there wont be with serverless either).
Now we just always resume the watcher straight after restarting, which shouldn't cause any problem

Reply:
Might need a quick call to review why this is fine.